src/pages/guest-home-page/GuestHome.jsx

```diff
@@ -13,33 +13,39 @@ import HowItWorks from '~/containers/guest-home-page/how-it-works/HowItWorks'
 import WhoWeAre from '~/containers/guest-home-page/who-we-are/WhoWeAre'
 import EmailConfirmModal from '~/containers/email-confirm-modal/EmailConfirmModal'
 import ResetPassword from '~/containers/guest-home-page/reset-password/ResetPassword'
+import Footer from "~/containers/layout/footer/Footer";
+import ScrollToTopButton from "~/components/scroll-to-top-button/ScrollToTopButton";
 
 const GuestHomePage = () => {
-  const { openModal } = useContext(ModalContext)
-  const [searchParams, setSearchParams] = useSearchParams()
-
-  useEffect(() => {
-    const confirmToken = searchParams.get('confirmToken')
-    const resetToken = searchParams.get('resetToken')
-    const login = searchParams.get('login')
-
-    confirmToken && openModal({ component: <EmailConfirmModal confirmToken={ confirmToken } openModal={ openModal } /> })
-    resetToken && openModal({ component: <ResetPassword openModal={ openModal } resetToken={ resetToken } /> })
-    login !== null && openModal({ component: <LoginDialog /> })
-
-    setSearchParams([])
-  }, [searchParams, setSearchParams, openModal])
-
-  return (
-    <Box data-testid='guestHome' sx={ { flex: 1 } }>
-      <Welcome />
-      <Box sx={ { maxWidth: '1128px', margin: '0 auto', overflowX: 'hidden' } }>
-        <FeatureBlock items={ descriptionTimes } />
-        <WhatCanYouDo />
-        <HowItWorks />
-        <WhoWeAre />
-      </Box>
-    </Box>
-  )
+    const {openModal} = useContext(ModalContext)
+    const [searchParams, setSearchParams] = useSearchParams()
+
+    const element = useRef()
```

Comment:
Is it really necessary to use useRef here?

Reply:
yes